pixeltable/catalog/catalog.py

```diff
@@ -422,6 +425,176 @@ def create_view(
+        # do not yet exist in the store, they will be created as anonymous system tables.
+        for ancestor_md in md[1:]:
+            ancestor_id = UUID(ancestor_md.tbl_md.tbl_id)
+            replica = Catalog.get().get_table_by_id(ancestor_id)
+            replica_path: Path
+            if replica is None:
+                # We've never seen this table before. Create a new anonymous system table for it.
+                replica_path = Path(f'_system.replica_{ancestor_id.hex}', allow_system_paths=True)
+            else:
+                # The table already exists in the catalog. The existing path might be a system path (if the table
+                # was created as an anonymous base table of some other table), or it might not (if it's a snapshot
+                # that was directly replicated by the user at some point). In either case, use the existing path.
+                replica_path = Path(replica._path(), allow_system_paths=True)
+
+            # Store the metadata; it could be a new version (in which case a new record will be created) or a
+            # known version (in which case the newly received metadata will be validated as identical).
+            self.__store_replica_md(replica_path, ancestor_md)
+
+        # Update the catalog (as a final step, after all DB operations completed successfully).
+        # Only the table being replicated is actually made visible in the catalog.
+        self._tbls[tbl_id] = self._load_tbl(tbl_id)
+        return self._tbls[tbl_id]
+
+    def __store_replica_md(self, path: Path, md: schema.FullTableMd) -> None:
+        _logger.info(f'Creating replica table at {path!r} with ID: {md.tbl_md.tbl_id}')
+        dir = self._get_schema_object(path.parent, expected=Dir, raise_if_not_exists=True)
+        assert dir is not None
+
+        conn = Env.get().conn
+        tbl_id = md.tbl_md.tbl_id
+
+        new_tbl_md: Optional[schema.TableMd] = None
+        new_version_md: Optional[schema.TableVersionMd] = None
+        new_schema_version_md: Optional[schema.TableSchemaVersionMd] = None
+
+        # We need to ensure that the table metadata in the catalog always reflects the latest observed version of
+        # this table. (In particular, if this is a base table, then its table metadata need to be consistent
+        # with the latest version of this table having a replicated view somewhere in the catalog.)
+        q: sql.Executable = sql.select(schema.Table.md).where(schema.Table.id == tbl_id)
+        existing_md_row = conn.execute(q).one_or_none()
+
+        if existing_md_row is None:
+            # No existing table, so create a new record.
+            # If this is a proper ancestor of the table being replicated, it might be the case that the current_version
+            # and/or current_schema_version records in the given TableMd are strictly greater than the versions found
+            # in the given TableVersionMd and/or TableSchemaVersionMd. If that's the case, we need to adjust them so
```

Comment:
Let's do that adjustment when we create the replica md.

Reply:
Done